frost-core/src/frost/keys.rs

```diff
@@ -28,6 +28,52 @@ use super::compute_lagrange_coefficient;
 pub mod dkg;
 pub mod repairable;
 
+/// Sum the commitments from all peers into a group commitment.
+#[cfg_attr(feature = "internals", visibility::make(pub))]
+pub(crate) fn compute_group_commitment<C: Ciphersuite>(
+    commitments: &[VerifiableSecretSharingCommitment<C>],
+) -> VerifiableSecretSharingCommitment<C> {
+    let mut group_commitment =
+        vec![CoefficientCommitment(<C::Group>::identity()); commitments[0].0.len()];
+    for commitment in commitments {
+        for i in 0..group_commitment.len() {
+            group_commitment[i] =
+                CoefficientCommitment(group_commitment[i].value() + commitment.0[i].value());
+        }
+    }
+    VerifiableSecretSharingCommitment(group_commitment)
+}
+
+/// Computes a verifying share for a peer given the group commitment.
+#[cfg_attr(feature = "internals", visibility::make(pub))]
+pub(crate) fn compute_verifying_share<C: Ciphersuite>(
+    group_commitment: &VerifiableSecretSharingCommitment<C>,
+    peer: Identifier<C>,
+) -> VerifyingShare<C> {
+    VerifyingShare::new(evaluate_vss(group_commitment, peer))
+}
+
+/// Computes the group public key given the group commitment.
+#[cfg_attr(feature = "internals", visibility::make(pub))]
+pub(crate) fn compute_public_key<C: Ciphersuite>(
+    group_commitment: &VerifiableSecretSharingCommitment<C>,
+) -> VerifyingKey<C> {
+    VerifyingKey::new(group_commitment.first().expect("valid commitments").value())
+}
+
+/// Computes the public key package given a list of commitments.
+#[cfg_attr(feature = "internals", visibility::make(pub))]
+pub(crate) fn compute_public_key_package<C: Ciphersuite>(
+    members: &BTreeSet<Identifier<C>>,
+    group_commitment: &VerifiableSecretSharingCommitment<C>,
+) -> PublicKeyPackage<C> {
+    let mut verifying_keys = HashMap::new();
+    for peer in members {
+        verifying_keys.insert(*peer, compute_verifying_share(group_commitment, *peer));
+    }
+    PublicKeyPackage::new(verifying_keys, compute_public_key(group_commitment))
+}
+
```

Comment:
Now that I understand it better the better api would probably be to implement `Add` for `VerifiableSecretSharingCommitment` and add `from_commitment` methods to the `VerifyingShare`, `VerifyingKey` and `PublicKeyPackage`.